### PR TITLE
Throw TypeErrors on invalid inputs to AnimationEffectTiming

### PR DESCRIPTION
**Pull request by alancutter**
This change adds TypeErrors to setters on AnimationEffectTiming for `iterationStart`, `iterations`, `duration` and `easing`.
Relevant spec: http://w3c.github.io/web-animations/#animationeffecttiming

### CODE REVIEW

File: src/timing-utilities.js [hunk 1]
```diff
@@ -74,12 +76,18 @@
       return this._fill;
     },
     set iterationStart(value) {
+      if (value < 0) {
```
Comment (suzyh) [inline src/timing-utilities.js]: Should this have an isNan check as well, like duration?

File: test/js/timing-utilities.js [hunk 2]
```diff
@@ -119,4 +118,24 @@ suite('timing-utilities', function() {
...
   });
+  test('TypeErrors', function() {
+    var timing = normalizeTimingInput({
+      iterationStart: 123,
+      iterations: 456,
+      duration: 789,
+      easing: 'ease',
+    });
+
+    assert.throws(function() { timing.iterationStart = -1; });
```
Comment (suzyh) [inline test/js/timing-utilities.js]: Consider testing more invalid values

Comment (alancutter) [inline test/js/timing-utilities.js]: Added NaN tests.

File: src/timing-utilities.js (hunk 1 above)
Comment (alancutter) [inline src/timing-utilities.js]: Added check, I realise now the NaN check is implicit in the IDL type.

File: test/js/timing-utilities.js [hunk 3]
```diff
@@ -33,18 +33,17 @@ suite('timing-utilities', function() {
     f = toTimingFunction('cubic-bezier(0, 1, 1, 0)');
     assert.closeTo(f(0.104), 0.392, 0.01);
 
-    function assertIsLinear(easing) {
-      var f = toTimingFunction(easing);
-      for (var i = 0; i <= 1; i += 0.1) {
-        assert.equal(f(i), i, easing + ' is linear');
-      }
+    function assertThrows(easing) {
```
Comment (suzyh) [inline test/js/timing-utilities.js]: To make this test easier to read, I think this function name should be more explicit about what it does. This is a bit long but maybe something like assertThrowsOnConversionToTimingFunction? assertInvalidEasingThrows?

Comment (alancutter) [inline test/js/timing-utilities.js]: Done.

**Comment by suzyh** [comment]
lgtm

**Pull request merged**